apisix/plugins/ip-restriction/init.lua

```diff
@@ -89,8 +89,13 @@ end
 
 function _M.restrict(conf, ctx)
     local block = false
-    local remote_addr = ctx.var.remote_addr
-
+    local remote_addr_arr = ctx.var.http_x_forwarded_for
```

Comment:
You can solve it with a global real_ip configuration:
https://github.com/apache/apisix/blob/3773f2626e5753f95cc040c27b59fd0cff264513/conf/config-default.yaml#L226-L230

Or a route-level solution, the `real-ip` plugin: https://apisix.apache.org/docs/apisix/plugins/real-ip